web/Services/AzureStorage.cs

```diff
@@ -0,0 +1,115 @@
+        Task DeleteFile(string varType, int id, string blobFilename);
+    }
+
+    public class AzureStorage : IAzureStorage
+    {
+        private readonly string _storageConnectionString;
+        private readonly string _baseContainerName;
+        private readonly ILogger<AzureStorage> _logger;
+
+        public AzureStorage(IConfiguration configuration, ILogger<AzureStorage> logger)
+        {
+            _storageConnectionString = configuration.GetValue<string>("BlobConnectionString");
+            _baseContainerName = configuration.GetValue<string>("BlobContainerName");
+            _logger = logger;
+        }
+
+        private string GetContainerName(string varType, int id)
+        {
+            return _baseContainerName + varType + id.ToString();
+        }
+
+        private async Task<BlobContainerClient> GetOrCreateContainerClient(string varType, int id)
+        {
+            string containerName = GetContainerName(varType, id);
+            BlobServiceClient blobServiceClient = new BlobServiceClient(_storageConnectionString);
+            var containerClient = blobServiceClient.GetBlobContainerClient(containerName);
+            if (!containerClient.Exists())
+            {
+                await blobServiceClient.CreateBlobContainerAsync(containerName);
+                containerClient = new BlobContainerClient(_storageConnectionString, containerName);
+            }
+            return containerClient;
+        }
+
+        public async Task DeleteContainer(string varType, int id)
+        {
+            var containerClient = await GetOrCreateContainerClient(varType, id);
+            await containerClient.DeleteAsync();
+        }
+
+        public async Task UploadFile(IFormFile blob, string varType, int id)
+        {
+            var containerClient = await GetOrCreateContainerClient(varType, id);
+            var blobClient = containerClient.GetBlobClient(blob.FileName);
+            int i = 0;
+            while (blobClient.Exists())
+            {
+                i += 1;
+                string fileName = SplitFileName(blob.FileName).name + i + "." + SplitFileName(blob.FileName).type;
+                blobClient = containerClient.GetBlobClient(fileName);
+            }
+            await using (Stream? data = blob.OpenReadStream())
+            {
+                await blobClient.UploadAsync(data);
+            }
+        }
+
+        public async Task<ImageListViewModel> ListFiles(string varType, int id)
+        {
+            var containerClient = await GetOrCreateContainerClient(varType, id);
+            List<string> fileNames = new List<string>();
+            await foreach (BlobItem file in containerClient.GetBlobsAsync())
+            {
+                fileNames.Add(file.Name);
+            }
+            ImageListViewModel imageList = new ImageListViewModel()
+            {
+                PropertyId = id,
+                FileList = fileNames
+            };
+            return imageList;

```

Comment:
Don't ned to change it, but as a minor point, usually Controllers will construct ViewModels rather than the service - the idea being the service method can then be reused easily between controllers rather than being tied to a particular page.